app/Jobs/GetDataFromKobo.php

```diff
@@ -136,76 +132,76 @@ public function handle()
-                        $cultivo['_id'] =  $newSubmission['_id'];
-
-                        // get the cultivo_id from the creation of thge cultivo
-                        $new_cultivo = DataMapController::newRecord($dataMap, $cultivo);
-                        $cultivo['cultivo_id'] =  $new_cultivo->id;
-
-                        $cultivo_modules = $cultivo['modulos_cultivo'] . ' '. $cultivo['modulo_cultivo_loop']['extra_modulo_cultivo'];
-                        $cultivo_modules = explode(' ', $cultivo_modules);
-                        $cultivo = $cultivo + $cultivo['modulo_cultivo_loop'];
-                        unset($cultivo['modulo_cultivo_loop']);
-
-                        foreach ($cultivo_modules as $cultivo_module) {
-
-                            if($cultivo_module == 3){
-                                if($cultivo['problema'] == 'plagas' || $cultivo['problema'] == 'ambas') {
-                                    $cultivo_module = 'plagas';
-                                    
-                                    foreach ($cultivo['plaga_repeat'] as $plaga) {
-                                        $dataMap_cultivo_module = DataMap::findorfail($cultivo_module);
-                                        $plaga['_id'] =  $newSubmission['_id'];
-                                        $plaga['cultivo_id'] =  $new_cultivo->id;
-                                        $new_module = DataMapController::newRecord($dataMap_cultivo_module, $plaga);
-                                    }
-                                }
-                                if($cultivo['problema'] == 'plagas' || $cultivo['enfermedades'] == 'ambas') {
-                                    $cultivo_module = 'enfermedades';
-                                    $dataMap_cultivo_module = DataMap::findorfail($cultivo_module);
-                                    $new_module = DataMapController::newRecord($dataMap_cultivo_module, $cultivo);  
-
-                                }
-
-                            } else {
-                                $dataMap_cultivo_module = DataMap::findorfail($cultivo_module);
-                                $new_module = DataMapController::newRecord($dataMap_cultivo_module, $cultivo);  
-                                
-                            }  
+                // Iterate through the other parcela-level modules.
+                foreach ($newSubmission['modulos'] as $parcelaModule) {
+                    if ($parcelaModule === 'C') {
+                        $this->processCultivoData($newSubmission);
+                    } else {
+                        $dataMap = DataMap::findOrFail($parcelaModule);
+                        $data = $newSubmission['modulo_loop'];
+                        $data['parcela_id'] = $newSubmission['parcela_id'];
+                        $data['_id'] = $newSubmission['_id'];
+
+                        DataMapController::newRecord($dataMap, $data);
+                    }
```

Comment:
Instead of the hard-coded check for module "A" (soils), I updated this to run through all plot-level modules. I left the hard-coded check for "C" (crops), but moved that section into a separate method for easier reading. 

This should now work for the suelo, muestra_suelo and any future plot-level modules that get added.